### ISSUE TITLE
1.4.4 Resize Text: Potential changes based on AG WG review

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#772]
In the AG WG review of the recent updates to WCAG2ICT (Issue #750) there was some concern expressed regarding the introductory language, and potentially also the notes. We'll cover this in 2 parts.
# 1. Introductory language
Possible changes to the introductory information on the applicability of [1.4.4 Resize Text](https://w3c.github.io/wcag2ict/#applying-sc-1-4-4-resize-text-to-non-web-documents-and-software). There have been two suggested changes to this language in issue 750.

## Proposal 0 - Current language in WCAG2ICT
>This success criterion is problematic to apply directly to non-web software because not all platforms provide text enlargement features that increase all displayed text to 200%. Non-web software needs to work with platform capabilities where they exist, but when the platform has text resizing that up to 200%, but not all text types scale to 200%, it is unreasonable for all apps on a particular platform to be required to build in their own text resizing. A reasonable expectation would be for non-web software to work with the text sizing features to the extent the platform provides. Doing so would still address the user needs identified in the [Intent from Understanding Success Criterion 1.4.4](https://www.w3.org/WAI/WCAG22/Understanding/resize-text#intent). The following criterion is recommended as a substitute for the WCAG language:

## Proposal 1 - from [Gregg's comment](https://github.com/w3c/wcag2ict/issues/750#issuecomment-3191761896)
Modifies most of the introductory text.
>This success criterion is problematic to apply directly to non-web software because not all platforms provide text enlargement features that increase all displayed text to 200%.  **Some platforms such as some mobile platforms do not provide built-in utilities to allow scaling of all text to 200% whereas all browsers did when the SC was written. The task force felt that the intent was to support scaling up to at least 200% by the platform, not to require that all software provide its own scaling function. However, if there is no scaling function in the platform at all then text scaling should be provided to ensure that text is large enough for those with 20/40 vision (the equivalent of 200% of 20/20). Note that this success criterion also has special considerations for ICT with closed functionality.** The following criterion is recommended as a substitute for the WCAG language:

## Proposal 2 - from [Jon's comment](https://github.com/w3c/wcag2ict/issues/750#issuecomment-3198940196)
Rewords the second from the last sentence starting with "A reasonable expectation..."

>This success criterion is problematic to apply directly to non-web software because not all platforms provide text enlargement features that increase all displayed text to 200%. Non-web software needs to work with platform capabilities where they exist, but when the platform has text resizing that up to 200%, but not all text types scale to 200%, it is unreasonable for all apps on a particular platform to be required to build in their own text resizing. A reasonable expectation would be for non-web software, **where the platform has text resizing support up to 200%, but where not all text resizes to 200% of it's default due to the resized text already being 200% of the default body text size, resizes** to the extent the platform provides. Doing so would still address the user needs identified in the [Intent from Understanding Success Criterion 1.4.4](https://www.w3.org/WAI/WCAG22/Understanding/resize-text#intent). The following criterion is recommended as a substitute for the WCAG language:

# 2. Possible adjustments to the notes
Jon expressed concern about the notes, but we had made some adjustments so it's difficult to tell what might need changing. We currently have 2 notes that are similar to what he commented on - Notes 3 and 4

>**Note 3 (Added) (for non-web software)** 
>The [Intent section in Understanding 1.4.4 Resize Text](http://section/) refers to the ability to allow users to enlarge the text on screen at least up to 200% without needing to use [assistive technologies](https://w3c.github.io/wcag2ict/#dfn-assistive-technologies). This means that the [non-web software](https://w3c.github.io/wcag2ict/#software) provides some means for enlarging the text 200% (zoom or otherwise) without loss of [content](https://w3c.github.io/wcag2ict/#content-on-and-off-the-web) or functionality, or that the non-web software works with the platform features to satisfy this success criterion.

>**Note 4 (Added) (for non-web software)**
For non-web software, sometimes the platform provides text scaling to 200% for most, but not all text (e.g. headings, which are naturally large, may not be increased in size to 200%, but other text does increase to 200%). In such cases, authors would only need to support text scaling to the extent provided by user settings in the platform, without losing content or functionality, to satisfy this success criterion.

Do we need both notes?  Should one be eliminated? Should the note(s) be edited further?

**Comment by maryjom** [comment]
@GreggVan @mraccess77 Just want to double check that this issue characterizes the concerns and proposals you both added in the AG WG Review issue. Jon, it did seem like you might also want adjustments to the notes - probably Note 3 - but I wasn't sure if you felt the notes also needed adjusting if the introductory text was more explanatory. Perhaps removing Note 3 would be reasonable since Note 4, I believe, provides the boundaries more clearly.

**Comment by bruce-usab** [comment]
1. I prefer Proposal 2 over Proposal 1.  I don’t agree with Proposal 0, for the concerns @mraccess77 raised.  I don't think Proposal 0 is warrants delay in publication.
2. I think we need both notes and that neither needs any more word smithing.

Whichever we settle upon (either Proposal 1 or 2) I will have minor editorial suggestions.  I will wait to provide that feedback, pending the preferred direction of the TF.    😄

**Comment by GreggVan** [comment]
- Proposal  0  doesn't address the issues
- Proposal 2.  Uses " A reasonable expectation would be "  which leaves one baffled as to whether it is required or we think it should be a recommendation not a requirements - or we recommend that as the requirement.

I think Proposal 1 is the best (and only one left standing.)



HOWEVER - let me put one more on the table. 

Proposal 3

Here is a variation on that -- that I  think covers all the angles including places where no resizing is needed to achieve the same thing as 200% increase over normal sized text.

This success criterion is problematic to apply directly to non-web software when platforms do not provide text enlargement features that increase all displayed text to 200%. For example some mobile platforms do not provide built-in utilities to allow scaling of all text to 200% whereas all browsers did when the SC was written - and the intent when the original SC was written was to support and not interfere with that (not require software to provide their own). However, if there is no scaling function in the platform at all, then text scaling should be provided to ensure that text is large enough for those with 20/40 vision (the equivalent of 200% of 20/20). Note that this success criterion also has special considerations for ICT with closed functionality. The following criterion is therefor recommended as a substitute for the WCAG language when applying to non-web software:

Except for captions and images of text one of the following is true
a) text can be resized without loss of content or functionality without assistive technology up to 200 percent using text resizing functionality of the platform or, if the platform provides text resizing capabilities but it does not reach 200 percent for all text, up to the text sizing capabilities of the platform.
b) if the non-web software is designed to work on a platform that has no text resizing capabilities, or that functionality is not exposed to the user when using the software, text can be resized without loss of content or functionality without assistive technology up to 200 percent using text resizing functionality of the software
c) software is running on a known hardware platform and all text is already known to be 3/8 of an in or larger in size.

**Comment by GreggVan** [comment]
I didnt think we were trying to write requirements for products with closed functionality.   Just noting the problems that might arise with certain provisions in the list of SC in the appendix related to closed products.

**Comment by GreggVan** [comment]
Good thought  with regard to TVs and Virtual displays.  For signage - it would not usually apply since the provision requires software with a human interface. 

I  opened an issue in gitlab saying adding to the end of that note  ".. for content expected to be viewed at ~400mm."

**Comment by maryjom** [comment]
Additional PR to remove the minimum text size note. This isn't interpretation of Resize Text, but something the EN felt they should have to address EAA requirement for readable text.
